Makefile.in

```diff
@@ -73,38 +89,56 @@ depend:
 	for i in $(SUBDIRS) doc example docker; \
 	do (cd $$i && $(MAKE) $(MFLAGS) depend); done
 
-install:  
+# Needs root permissions on most systems by default
+install: checkroot
 	for i in $(SUBDIRS) doc; \
 	do (cd $$i; $(MAKE) $(MFLAGS) $@)||exit 1; done; 
 	$(MAKE) $(MFLAGS) install-include
 
 install-include:
 	for i in $(SUBDIRS) doc; \
 	do (cd $$i && $(MAKE) $(MFLAGS) $@)||exit 1; done; 
-	echo "To install example app: cd example; make; make install"
+	@echo "\e[32mTo install example app and test utils: make install-example\e[0m"
 
-uninstall:
-	for i in $(SUBDIRS) doc example docker; \
+uninstall: checkroot
+	for i in $(SUBDIRS) doc example util docker; \
 	do (cd $$i && $(MAKE) $(MFLAGS) $@)||exit 1; done; 
 
-doc:	
+doc: warnroot
 	cd $@; $(MAKE) $(MFLAGS) $@
 
+util:
+	cd $@; $(MAKE) $(MFLAGS)
+
+clean-example:
+	for i in example util; \
+	  do (cd $$i && $(MAKE) $(MFLAGS) clean) || exit 1; done;
+
+install-example: checkroot
+	for i in example util; \
+	  do (cd $$i && $(MAKE) $(MFLAGS) install) || exit 1; done;
+
+uninstall-example: checkroot
+	for i in example util; \
+	  do (cd $$i && $(MAKE) $(MFLAGS) uninstall) || exit 1; done;
+
 config.status:	configure
 	$(SHELL) config.status --recheck
 
-configure:	configure.ac
+configure: configure.ac
 	cd $(srcdir) && autoconf
 
 clean:
-	for i in $(SUBDIRS) doc example docker; \
+	for i in $(SUBDIRS) doc example util docker; \
 		do (cd $$i && $(MAKE) $(MFLAGS) $@); done; 
 
+mrproper: uninstall uninstall-example clean clean-example
```

Comment:
Please add a comment: what does / why use, mrproper rule?

Reply:
Added a comment in latest commit